### DISCUSSION TITLE
Roadmap on integrating community-templates into Netbox natively

### DISCUSSION THREAD

**Discussion by AliMickey** [netbox-community/netbox#16729] [General]
Hi,

I have noticed since end of 2023 that the device-types repository has been brought under the official Netbox org and stated "We will work to get this fully supported soon.".

Is there anything being done to get native syncing of templates without the use of the import tool (which currently does not work on Netbox 4.0.x). 

Thanks

**Comment by candlerb** [comment]
> Is there anything being done to get native syncing of templates without the use of the import tool (which currently does not work on Netbox 4.0.x)

Which "import tool" "does not work"? Can you give an example to reproduce? What *exact* version of 4.0.x are you running?

**Reply by AliMickey** [reply]
> > Is there anything being done to get native syncing of templates without the use of the import tool (which currently does not work on Netbox 4.0.x)
> 
> Which "import tool" "does not work"? Can you give an example to reproduce? What _exact_ version of 4.0.x are you running?

Hi @candlerb,

I'm on 4.0.5 currently.

This thread is an example https://github.com/netbox-community/Device-Type-Library-Import/issues/134

**Reply by candlerb** [reply]
OK, so you're talking specifically about the script `nb-dt-import.py` from https://github.com/netbox-community/Device-Type-Library-Import which seems pretty old: the front page says "Tested working with 2.9.4, 2.10.4"

I haven't come across this problem because I simply copy-paste the YAML from the device type library into the Netbox web UI. I can see this would be annoying if you have lots of different device types to import.

**Comment by jeremystretch** [comment]
No work is currently underway. Is this something you'd like to volunteer to work on @AliMickey?

**Reply by AliMickey** [reply]
Unfortunately getting this integrated natively within the UI is out of my skill scope, I am currently working on making a simpler script that will also support updating existing device instances of a changed template which I will publish, but not quite as streamlined as a GUI button.

**Reply by candlerb** [reply]
> I am currently working on making a simpler script that will also support updating existing device instances of a changed template

Aside: you might get some inspiration from [this script](https://github.com/netbox-community/customizations/blob/master/scripts/add_device_type_components.py), if you've not seen it already.

That script only adds new components; it does not modify or remove components which are not in the device type template. Note also that there are a couple of outstanding pull requests: [#101](https://github.com/netbox-community/customizations/pull/101), [#107](https://github.com/netbox-community/customizations/pull/107)
